server/src/main/java/org/elasticsearch/action/admin/indices/alias/get/GetAliasesRequest.java

```diff
@@ -27,20 +28,18 @@
 import org.elasticsearch.common.io.stream.StreamOutput;
 
 import java.io.IOException;
+import java.util.Objects;
 
 public class GetAliasesRequest extends MasterNodeReadRequest<GetAliasesRequest> implements AliasesRequest {
 
     private String[] indices = Strings.EMPTY_ARRAY;
     private String[] aliases = Strings.EMPTY_ARRAY;
-
     private IndicesOptions indicesOptions = IndicesOptions.strictExpand();
+    private boolean aliasesProvided = false;
```

Comment:
I keep on having problems remembering what `aliasesProvided` means as with either `aliases` or `setAliases` aliases can be specified ...
What about adding some comments to the `aliases` and `setAliases` to make the difference more obvious to the users of this class ?